activerecord/test/cases/connection_pool_test.rb

```diff
@@ -772,5 +789,88 @@ def with_single_connection_pool
           pool.disconnect! if pool
         end
     end
+
+    class ConnectionPoolThreadTest < ActiveRecord::TestCase
+      include ConnectionPoolTests
+
+      class ThreadConnectionTestModel < ActiveRecord::Base
+        self.abstract_class = true
+      end
+
+      def setup
+        super
+        shared_setup(ThreadConnectionTestModel, :thread)
+      end
+
+      teardown do
+        shared_teardown
+      end
+
+      private
+        def new_thread(*args, &block)
+          Thread.new(*args, &block)
+        end
+
+        def stop_thread
+          Thread.stop
+        end
+
+        def pass_to(_thread)
+          Thread.pass
+        end
+
+        def kill_current
+          Thread.current.kill
+        end
```

Comment:
It's a bit handwavy, but I'm scared that this may be called in the main thread causing minitest to exit 0 and to silently skip a large part of the test suite. Could we install call kill on the thread stored in a variable or received as argument instead? That would make me sleep better.